influxdb/metric.go

```diff
@@ -147,10 +145,46 @@ func (m *Metric) metricDescriptor() *metricpb.MetricDescriptor {
 	}
 }
 
+func (m *Metric) filterPoints(points []point, endTime time.Time) ([]point, error) {
+	if !m.config.TimeAggregated {
+		// For non-time-aggregated gauge metrics, we don't need to apply any
+		// filtering since we only query InfluxDB for new points, and points
+		// that are too fresh are ignored by applying an offset.
+		return points, nil
+	}
+
+	queryInterval, err := m.config.queryInterval()
+	if err != nil {
+		return nil, err
+	}
+
+	var filteredPoints []point
+	for _, p := range points {
+		// Once cumulative metrics come in play, we'll filter using lastPoint.
+		// For now we can rely on our constructed query to only fetch the
+		// latest points.
+
+		// For time-aggregated InfluxQL queries with interval i, Influx returns
+		// points with timestamp x, denoting the aggregated value from time
+		// [i, i + x).
+		intervalEnd := p.timestamp.Add(queryInterval)
```

Comment:
Instead of throwing away `intervalEnd`, and later separately parsing and adding `queryInterval` to each point (in `convertPoint`), should we just overwrite `p.timestamp` here?

Reply:
Good idea, added.